javascript_nodejs/10.prompt-validations/dialogs/onboard/index.js

```diff
@@ -0,0 +1,114 @@
+        this.addDialog(new WaterfallDialog(START_DIALOG,[
+            // If a user name is already set, switch to the HELLO_USER dialog.
+            // Otherwise, continue with collecting values from the user.
+            async (dc, step) => {
+                const user = await userProfile.get(dc.context, {});
+                if (user.name) {
+                    return dc.replace(HELLO_USER);
+                } else {
+                    return await dc.prompt(GET_NAME_PROMPT, `What is your name, human?`);
+                }
+            },
+            // Collect user name, then prompt for age. 
+            async (dc, step) => {
+                step.values[USER_NAME_PROPERTY] = step.result;
+                return await dc.prompt(GET_AGE_PROMPT, `What is your age?`);
+            },
+            // Collect age, then prompt for date of birth.
+            async (dc, step) => {
+                step.values[AGE_PROPERTY] = step.result;
+                return await dc.prompt(GET_DOB_PROMPT, `What is your date of birth?`);
+            },
+            // Collect date of birth, then prompt for favorite color.
+            async (dc, step) => {
+                step.values[DOB_PROPERTY] = step.result;
+                const choices = ['red','blue','green'];
+                return await dc.prompt(GET_COLOR_PROMPT, `Finally, what is your favorite color?`, choices);
+            },
+            // Collect favorite color and continue.
+            async (dc, step) => {
+                step.values[COLOR_PROPERTY] = step.result;
+                return await step.next();
+            },
+            // With all values in hand, we can now store them in our model and complete.
+            async (dc, step) => {
+
+                const user = await userProfile.get(dc.context, {});
+                user[USER_NAME_PROPERTY] = step.values[USER_NAME_PROPERTY];
+                user[AGE_PROPERTY] =  step.values[AGE_PROPERTY];
+                user[DOB_PROPERTY] =  step.values[DOB_PROPERTY];
+                user[COLOR_PROPERTY] =  step.values[COLOR_PROPERTY];
+                await userProfile.set(dc.context, user);
+        
+                await dc.context.sendActivity(`Your profile is complete! Thank you.`);
+
+                // Transition to the display of the profile data.
+                return await dc.begin(HELLO_USER);
+            }
+        ]));
+
+        // This dialog loads and displays the information previously provided by the user.
+        this.addDialog(new WaterfallDialog(HELLO_USER, [
+            async (dc, step) => {
+                const user = await userProfile.get(dc.context, {});
+
+                const text = [
+                    `You asked me to call you ${user[USER_NAME_PROPERTY]}.`,
+                    `You were born on ${ moment( user[DOB_PROPERTY] ).format("MMM Do, YYYY") } and claim to be ${ user[AGE_PROPERTY] }.`,
+                    `Your favorite color is ${ user[COLOR_PROPERTY] }.`
```

Comment:
The spacing for template literals is not consistent for lines 89 through 91. Please remove the spacing from the `moment()` on line 90 call and add spacing to line 89.